src/v/storage/tests/log_segment_appender_test.cc

```diff
@@ -31,7 +31,7 @@ SEASTAR_THREAD_TEST_CASE(test_can_append_multiple_flushes) {
                  | ss::open_flags::truncate)
                .get0();
     auto appender = segment_appender(
-      f, segment_appender::options(ss::default_priority_class(), 1));
+      f, segment_appender::options(ss::default_priority_class(), 1, 16_KiB));
```

Comment:
this looks good, but can we parameter all the existing tests, not just a new test that is added, because presumably all of the existing functionality should continue to be preserved independent of the allocation size.

I think factoring out the body of the tests into helper functions will make this straight forward. Something like:

```
static void test_can_append_multiple_flushes(size_t size) {
   ...
}

SEASTAR_THREAD_TEST_CASE(test_can_append_multiple_flushes) {
   run_test_can_append_multiple_flushes(16_KiB);
   run_test_can_append_multiple_flushes(32_MiB);
```